mucommander-core/src/main/java/com/mucommander/ui/action/impl/ShowFilePopupMenuAction.java

```diff
@@ -23,46 +23,55 @@
 
 import javax.swing.KeyStroke;
 
-import com.mucommander.commons.file.util.FileSet;
+import com.mucommander.commons.file.AbstractFile;
+import com.mucommander.core.desktop.DesktopManager;
+import com.mucommander.text.Translator;
 import com.mucommander.ui.action.AbstractActionDescriptor;
 import com.mucommander.ui.action.ActionCategory;
 import com.mucommander.ui.action.ActionDescriptor;
+import com.mucommander.ui.action.ActionProperties;
+import com.mucommander.ui.dialog.InformationDialog;
 import com.mucommander.ui.main.MainFrame;
 import com.mucommander.ui.main.menu.TablePopupMenu;
 import com.mucommander.ui.main.table.FileTable;
+import com.mucommander.ui.main.table.FileTableModel;
 
 
 /**
- * This action shows the popup menu for currently selected file
- *
- * @author Miroslav Oujesky
+ * This action shows the popup menu for currently selected file.
+ * Exactly the same popup menu appears also when right-clicking on file/folder.
  */
-public class ShowFilePopupMenuAction extends SelectedFilesAction {
+public class ShowFilePopupMenuAction extends SelectedFileAction {
 
     public ShowFilePopupMenuAction(MainFrame mainFrame, Map<String, Object> properties) {
         super(mainFrame, properties);
+        setEnabled(DesktopManager.canOpenInFileManager());
     }
 
     @Override
-    public void performAction(FileSet files) {
-        FileTable table = mainFrame.getActiveTable();
-
-        int currentRow = table.getSelectedRow();
-
-        // Does the row correspond to the parent '..' folder ?
-        boolean parentFolderClicked = currentRow == 0 && table.getFileTableModel().hasParentFolder();
-
-        TablePopupMenu popupMenu = new TablePopupMenu(
-            mainFrame,
-            table.getFolderPanel().getCurrentFolder(),
-            parentFolderClicked ? null : table.getSelectedFile(),
-            parentFolderClicked,
-            files
-        );
+    public void performAction() {
+        AbstractFile selectedFile = mainFrame.getActiveTable().getSelectedFile();
+
+        try {
+            FileTable fileTable = mainFrame.getActiveTable();
+            FileTableModel tableModel = (FileTableModel) fileTable.getModel();
+            int selectedRow = fileTable.getSelectedRow();
+            // column 1 - where the name is (0 is icon) - is there any way to avoid hardcoding?
```

Comment:
yes, see [this](https://github.com/mucommander/mucommander/blob/5e4af240b6a2b5b5968c6ba9ddd20cdabcf16b01/mucommander-core/src/main/java/com/mucommander/ui/main/table/FileTable.java#L1014)

Reply:
thx